### PR TITLE
NanoUI surgery

### PR DESCRIPTION
**Pull request by ACCount12**
This was a nice little refactor, 20 minutes in and out.

New surgery interface:
изображение
To open this UI, you need apply scalpel and retractor to an organic body part, as usual, or use a screwdriver on a robotic BP. Closing an incision requires cautery, closing an open robot limb requires applying screwdriver again.

## Changelog
:cl: ACCount
tweak: The surgery system was remade entirely.
add: You can now operate on yourself by strapping yourself to a chair.
fix: You no longer need to have alt-tab to wiki all the tine just to operate on people.
fix: You can no longer pull a cruciform out by an accident. Unless you fuck up big time.
del: Internal bleeding is gone. Play crab rave.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/tools/scalpels.dm [hunk 1]
```diff
@@ -32,12 +32,17 @@
...
+	tool_qualities = list(QUALITY_CUTTING = 40, QUALITY_WIRE_CUTTING = 20, QUALITY_LASER_CUTTING = 40)
 	degradation = 0.11
 	use_power_cost = 0.12
 	suitable_cell = /obj/item/weapon/cell/small
 	max_upgrades = 4
 
+// Laser cutting overrides normal cutting
+/obj/item/weapon/tool/scalpel/laser/get_tool_type(mob/living/user, list/required_qualities, atom/use_on, datum/callback/CB)
+	if(QUALITY_LASER_CUTTING in required_qualities)
+		required_qualities -= QUALITY_CUTTING
```
Comment (LethalGhost) [inline code/game/objects/items/weapons/tools/scalpels.dm]: Is LASER CUTTING sill a thing without surgery steps?

File: code/modules/organs/internal/_internal.dm [hunk 2]
```diff
@@ -48,9 +47,56 @@
...
 		return
 	if(owner.chem_effects[CE_BLOODCLOT])
 		amount *= 1 + owner.chem_effects[CE_BLOODCLOT]
-	damage = between(0, damage - round(amount, 0.1), max_damage)
\ No newline at end of file
+	damage = between(0, damage - round(amount, 0.1), max_damage)
+
+
+// Gets the limb this organ is located in, if any
+/obj/item/organ/internal/proc/get_limb()
```
Comment (LethalGhost) [inline code/modules/organs/internal/_internal.dm]: AFAIR, that organ is already hold in `parent` var.

File: code/game/objects/items/weapons/tools/scalpels.dm (hunk 1 above)
Comment (ACCount12) [reply code/game/objects/items/weapons/tools/scalpels.dm]: Yes, you still use it when opening up a body part.

File: code/modules/organs/internal/_internal.dm (hunk 2 above)
Comment (ACCount12) [reply code/modules/organs/internal/_internal.dm]: That's only true for `external` organs, and I'm not about to start tampering with that code. At least not yet.

**Pull request merged**